blog/2023-02-09-http-s3-blog/index.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Persistent Log Stream with Amazon S3 Bucket
+description: We will learn how to build a Persistent Log Stream to store logs requests made to a Web Server in an Amazon S3 Bucket
+type: blog
+category: tutorial
+layout: single
+lastmod: 2023-02-20
+author: Ehis
+tags: ["S3", "AWS", "HTTP", "Real-time"]
+---
+
+## Table of Contents
+
+- [Introduction](#introduction)
+  - [What is Amazon S3?](#what-is-amazon-s3)
+  - [What is HTTP Request?](#what-is-http-request)
+- [Pre-requisite](#pre-requisite)
+- [How to Log HTTP Requests to S3 Bucket](#how-to-log-http-requests-to-s3-bucket)
+  - [Step 1: Deploy Vanus on the Playground](#step-1-deploy-vanus-on-the-playground)
+  - [Step 2: Make directory for S3 Sink Connector and Create Config file](#step-2-make-directory-for-s3-sink-connector-and-create-config-file)
+  - [Step 3: Make directory for HTTP Source Connector and Create Config file](#step-3-make-directory-for-http-source-connector-and-create-config-file)
+  - [Step 4: Create Subscription](#step-4-create-subscription)
+  - [Step 5: Make a request using CURL](#step-5-make-a-request-using-curl)
+- [Check out the result](#check-out-the-result)
+- [Conclusion](#conclusion)
+
+## Introduction
+
+Storing Logs on your web server may seem pretty okay for Low traffic websites like a personal blog but what about an e-commerce website which is a good example of a website that can receive millions of requests in a day? Storing such a high volume of logs may lead to more resources being needed to handle such logs. Also, the log files cannot be accessed if there is an issue with the server. So then, what can we do? Before we begin, I will explain some terminologies to us.
```

Comment:
```suggestion
When it comes to low-traffic websites, storing logs on the web server may not cause any issues. However, for high-traffic websites such as e-commerce sites that receive millions of requests per day, storing such a massive amount of logs can pose some challenges. Firstly, it can require more resources to handle the logs, which can increase the cost of maintaining the website. Additionally, if there is a problem with the server, the log files may not be accessible, which can make troubleshooting difficult.
```

Reply:
Done